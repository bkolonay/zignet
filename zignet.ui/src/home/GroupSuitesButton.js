import React, { Component } from 'react';

class GroupSuitesButton extends Component {

  render() {
    return (
      <a href="/?group=true" className="btn btn-primary" role="button" title="Click to group results by environment">Group</a>
    );
  }
}

export default GroupSuitesButton;
