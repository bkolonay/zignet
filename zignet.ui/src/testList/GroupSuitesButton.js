import React, { Component } from 'react';

class GroupSuitesButton extends Component {

  render() {
  	const suiteId = this.props.suiteId;
    const button = this.props.grouped ? 
      <a href={"/" + suiteId} className="btn btn-outline-primary float-right" role="button" title="Click to ungroup results">Ungroup</a> :
      <a href={"/" + suiteId + "?group=true"} className="btn btn-primary float-right" role="button" title="Click to group tests by environment">Group</a>;

    return (
      button
    );
  }
}

export default GroupSuitesButton;
