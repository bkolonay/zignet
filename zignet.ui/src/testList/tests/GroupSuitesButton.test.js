import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import GroupSuitesButton from '../GroupSuitesButton';

it('renders when not grouped', () => {
  const component = <GroupSuitesButton grouped={false} suiteId={1} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('renders when not grouped', () => {
  const component = <GroupSuitesButton grouped={true} suiteId={0} />;
  ReactDOM.render(component,document.createElement('div'));
});

it('snapshot when not grouped', () => {
  const component = <GroupSuitesButton grouped={false} suiteId={1} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshot when grouped', () => {
  const component = <GroupSuitesButton grouped={true} suiteId={0} />;
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});