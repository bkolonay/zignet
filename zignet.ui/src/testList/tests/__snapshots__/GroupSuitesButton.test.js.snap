// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot when grouped 1`] = `
<a
  className="btn btn-outline-primary float-right"
  href="/0"
  role="button"
  title="Click to ungroup results"
>
  Ungroup
</a>
`;

exports[`snapshot when not grouped 1`] = `
<a
  className="btn btn-primary float-right"
  href="/1?group=true"
  role="button"
  title="Click to group tests by environment"
>
  Group
</a>
`;
